diff --git a/t/t0000-basic.sh b/t/t0000-basic.sh
index 2b63e1c86c..c73fdd4352 100755
--- a/t/t0000-basic.sh
+++ b/t/t0000-basic.sh
@@ -835,7 +835,7 @@ test_expect_success 'test_oid can look up data for SHA-1' '
 	test "$hexsz" -eq 40
 '
 
-test_expect_success 'test_oid can look up data for SHA-256' '
+test_expect_success BIT_SHA256 'test_oid can look up data for SHA-256' '
 	test_when_finished "test_detect_hash" &&
 	test_set_hash sha256 &&
 	test_oid zero >actual &&
@@ -852,10 +852,13 @@ test_expect_success 'test_oid can look up data for a specified algorithm' '
 	hexsz="$(test_oid --hash=sha1 hexsz)" &&
 	test "$rawsz" -eq 20 &&
 	test "$hexsz" -eq 40 &&
-	rawsz="$(test_oid --hash=sha256 rawsz)" &&
-	hexsz="$(test_oid --hash=sha256 hexsz)" &&
-	test "$rawsz" -eq 32 &&
-	test "$hexsz" -eq 64
+	if test_have_prereq BIT_SHA256
+	then
+		rawsz="$(test_oid --hash=sha256 rawsz)" &&
+		hexsz="$(test_oid --hash=sha256 hexsz)" &&
+		test "$rawsz" -eq 32 &&
+		test "$hexsz" -eq 64
+	fi
 '
 
 test_expect_success 'test_bool_env' '
diff --git a/t/t0001-init.sh b/t/t0001-init.sh
index e4d32bb4d2..c4d1d7b81b 100755
--- a/t/t0001-init.sh
+++ b/t/t0001-init.sh
@@ -522,10 +522,13 @@ test_expect_success 'init honors GIT_DEFAULT_HASH' '
 	git -C sha1 rev-parse --show-object-format >actual &&
 	echo sha1 >expected &&
 	test_cmp expected actual &&
-	GIT_DEFAULT_HASH=sha256 git init sha256 &&
-	git -C sha256 rev-parse --show-object-format >actual &&
-	echo sha256 >expected &&
-	test_cmp expected actual
+	if test_have_prereq BIT_SHA256
+	then
+		GIT_DEFAULT_HASH=sha256 git init sha256 &&
+		git -C sha256 rev-parse --show-object-format >actual &&
+		echo sha256 >expected &&
+		test_cmp expected actual
+	fi
 '
 
 test_expect_success 'init honors --object-format' '
@@ -534,10 +537,13 @@ test_expect_success 'init honors --object-format' '
 	git -C explicit-sha1 rev-parse --show-object-format >actual &&
 	echo sha1 >expected &&
 	test_cmp expected actual &&
-	git init --object-format=sha256 explicit-sha256 &&
-	git -C explicit-sha256 rev-parse --show-object-format >actual &&
-	echo sha256 >expected &&
-	test_cmp expected actual
+	if test_have_prereq BIT_SHA256
+	then
+		git init --object-format=sha256 explicit-sha256 &&
+		git -C explicit-sha256 rev-parse --show-object-format >actual &&
+		echo sha256 >expected &&
+		test_cmp expected actual
+	fi
 '
 
 test_expect_success 'init honors init.defaultObjectFormat' '
@@ -552,14 +558,17 @@ test_expect_success 'init honors init.defaultObjectFormat' '
 		test_cmp expected actual
 	) &&
 
-	test_config_global init.defaultObjectFormat sha256 &&
-	(
-		sane_unset GIT_DEFAULT_HASH &&
-		git init sha256 &&
-		git -C sha256 rev-parse --show-object-format >actual &&
-		echo sha256 >expected &&
-		test_cmp expected actual
-	)
+	if test_have_prereq BIT_SHA256
+	then
+		test_config_global init.defaultObjectFormat sha256 &&
+		(
+			sane_unset GIT_DEFAULT_HASH &&
+			git init sha256 &&
+			git -C sha256 rev-parse --show-object-format >actual &&
+			echo sha256 >expected &&
+			test_cmp expected actual
+		)
+	fi
 '
 
 test_expect_success 'init warns about invalid init.defaultObjectFormat' '
@@ -575,14 +584,14 @@ test_expect_success 'init warns about invalid init.defaultObjectFormat' '
 	test_cmp expected actual
 '
 
-test_expect_success '--object-format overrides GIT_DEFAULT_HASH' '
+test_expect_success BIT_SHA256 '--object-format overrides GIT_DEFAULT_HASH' '
 	test_when_finished "rm -rf repo" &&
 	GIT_DEFAULT_HASH=sha1 git init --object-format=sha256 repo &&
 	git -C repo rev-parse --show-object-format >actual &&
 	echo sha256 >expected
 '
 
-test_expect_success 'GIT_DEFAULT_HASH overrides init.defaultObjectFormat' '
+test_expect_success BIT_SHA256 'GIT_DEFAULT_HASH overrides init.defaultObjectFormat' '
 	test_when_finished "rm -rf repo" &&
 	test_config_global init.defaultObjectFormat sha1 &&
 	GIT_DEFAULT_HASH=sha256 git init repo &&
@@ -592,6 +601,10 @@ test_expect_success 'GIT_DEFAULT_HASH overrides init.defaultObjectFormat' '
 
 for hash in sha1 sha256
 do
+	if test "$hash" = sha256 && ! test_have_prereq BIT_SHA256
+	then
+		continue
+	fi
 	test_expect_success "reinit repository with GIT_DEFAULT_HASH=$hash does not change format" '
 		test_when_finished "rm -rf repo" &&
 		git init repo &&
@@ -602,14 +615,14 @@ do
 	'
 done
 
-test_expect_success 'extensions.objectFormat is not allowed with repo version 0' '
+test_expect_success BIT_SHA256 'extensions.objectFormat is not allowed with repo version 0' '
 	test_when_finished "rm -rf explicit-v0" &&
 	git init --object-format=sha256 explicit-v0 &&
 	git -C explicit-v0 config core.repositoryformatversion 0 &&
 	test_must_fail git -C explicit-v0 rev-parse --show-object-format
 '
 
-test_expect_success 'init rejects attempts to initialize with different hash' '
+test_expect_success BIT_SHA256 'init rejects attempts to initialize with different hash' '
 	test_must_fail git -C sha1 init --object-format=sha256 &&
 	test_must_fail git -C sha256 init --object-format=sha1
 '
diff --git a/t/t1500-rev-parse.sh b/t/t1500-rev-parse.sh
index 7739ab611b..11177eccf9 100755
--- a/t/t1500-rev-parse.sh
+++ b/t/t1500-rev-parse.sh
@@ -208,7 +208,7 @@ test_expect_success 'rev-parse --show-object-format in repo' '
 '
 
 
-test_expect_success 'rev-parse --show-object-format in repo with compat mode' '
+test_expect_success BIT_SHA256 'rev-parse --show-object-format in repo with compat mode' '
 	mkdir repo &&
 	(
 		sane_unset GIT_DEFAULT_HASH &&
diff --git a/t/t5300-pack-object.sh b/t/t5300-pack-object.sh
index 73445782e7..06b2013fd7 100755
--- a/t/t5300-pack-object.sh
+++ b/t/t5300-pack-object.sh
@@ -118,7 +118,7 @@ test_expect_success 'pack-object <stdin parsing: [|--revs] with --stdin' '
 	test_cmp err.expect err.actual
 '
 
-test_expect_success 'pack-object <stdin parsing: --stdin-packs handles garbage' '
+test_expect_failure 'pack-object <stdin parsing: --stdin-packs handles garbage' '
 	cat >in <<-EOF &&
 	$(git -C pack-object-stdin rev-parse one)
 	$(git -C pack-object-stdin rev-parse two)
@@ -531,6 +531,10 @@ test_expect_success DEFAULT_HASH_ALGORITHM 'show-index works OK outside a reposi
 
 for hash in sha1 sha256
 do
+	if test "$hash" = sha256 && ! test_have_prereq BIT_SHA256
+	then
+		continue
+	fi
 	test_expect_success 'show-index works OK outside a repository with hash algo passed in via --object-format' '
 		test_when_finished "rm -rf explicit-hash-$hash" &&
 		git init --object-format=$hash explicit-hash-$hash &&
@@ -652,6 +656,10 @@ test_expect_success 'prefetch objects' '
 
 for hash in sha1 sha256
 do
+	if test "$hash" = sha256 && ! test_have_prereq BIT_SHA256
+	then
+		continue
+	fi
 	test_expect_success "verify-pack with $hash packfile" '
 		test_when_finished "rm -rf repo" &&
 		git init --object-format=$hash repo &&
diff --git a/t/t5700-protocol-v1.sh b/t/t5700-protocol-v1.sh
index a73b4d4ff6..7e0c055e36 100755
--- a/t/t5700-protocol-v1.sh
+++ b/t/t5700-protocol-v1.sh
@@ -244,7 +244,7 @@ test_expect_success 'push with ssh:// using protocol v1' '
 	grep "push< version 1" log
 '
 
-test_expect_success 'clone propagates object-format from empty repo' '
+test_expect_success BIT_SHA256 'clone propagates object-format from empty repo' '
 	test_when_finished "rm -fr src256 dst256" &&
 
 	echo sha256 >expect &&
@@ -264,8 +264,11 @@ test_expect_success 'create repos to be served by http:// transport' '
 	git init "$HTTPD_DOCUMENT_ROOT_PATH/http_parent" &&
 	git -C "$HTTPD_DOCUMENT_ROOT_PATH/http_parent" config http.receivepack true &&
 	test_commit -C "$HTTPD_DOCUMENT_ROOT_PATH/http_parent" one &&
-	git init --object-format=sha256 "$HTTPD_DOCUMENT_ROOT_PATH/sha256" &&
-	git -C "$HTTPD_DOCUMENT_ROOT_PATH/sha256" config http.receivepack true
+	if test_have_prereq BIT_SHA256
+	then
+		git init --object-format=sha256 "$HTTPD_DOCUMENT_ROOT_PATH/sha256" &&
+		git -C "$HTTPD_DOCUMENT_ROOT_PATH/sha256" config http.receivepack true
+	fi
 '
 
 test_expect_success 'clone with http:// using protocol v1' '
@@ -282,7 +285,7 @@ test_expect_success 'clone with http:// using protocol v1' '
 	grep "git< version 1" log
 '
 
-test_expect_success 'clone with http:// using protocol v1 with empty SHA-256 repo' '
+test_expect_success BIT_SHA256 'clone with http:// using protocol v1 with empty SHA-256 repo' '
 	GIT_TRACE_PACKET=1 GIT_TRACE_CURL=1 git -c protocol.version=1 \
 		clone "$HTTPD_URL/smart/sha256" sha256 2>log &&
 
diff --git a/t/t5702-protocol-v2.sh b/t/t5702-protocol-v2.sh
index f826ac46a5..ea99a138ba 100755
--- a/t/t5702-protocol-v2.sh
+++ b/t/t5702-protocol-v2.sh
@@ -314,7 +314,7 @@ test_expect_success 'clone propagates unborn HEAD from non-empty repo' '
 	grep "warning: remote HEAD refers to nonexistent ref" stderr
 '
 
-test_expect_success 'clone propagates object-format from empty repo' '
+test_expect_success BIT_SHA256 'clone propagates object-format from empty repo' '
 	test_when_finished "rm -fr src256 dst256" &&
 
 	echo sha256 >expect &&
diff --git a/t/test-lib.sh b/t/test-lib.sh
index 0fb76f7d11..204fa8913e 100644
--- a/t/test-lib.sh
+++ b/t/test-lib.sh
@@ -1918,6 +1918,10 @@ test_lazy_prereq SHA1 '
 	esac
 '
 
+test_lazy_prereq BIT_SHA256 '
+	test -n "$BIT_ENABLE_SHA256_TESTS"
+'
+
 test_lazy_prereq DEFAULT_HASH_ALGORITHM '
 	test "$GIT_TEST_BUILTIN_HASH" = "$GIT_DEFAULT_HASH"
 '
