diff --git a/t/t5610-clone-detached.sh b/t/t5610-clone-detached.sh
index a7ec21eda5..b4c9e31f00 100755
--- a/t/t5610-clone-detached.sh
+++ b/t/t5610-clone-detached.sh
@@ -31,8 +31,8 @@ test_expect_success 'cloned HEAD matches' '
 	git --git-dir=detached-branch/.git log -1 --format=%s >actual &&
 	test_cmp expect actual
 '
-test_expect_failure 'cloned HEAD is detached' '
-	head_is_detached detached-branch
+test_expect_success 'cloned HEAD is detached' '
+	head_is_detached detached-branch
 '

 test_expect_success 'clone repo (detached HEAD points to tag)' '
